### PR TITLE
Request form

### PR DESCRIPTION
**Pull request by berndvdveen**

### CODE REVIEW

File: podium-gateway/src/main/webapp/app/request/form/request-form.component.html [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+    <div class="col-md-6 offset-md-3">
+        <h1 *ngIf="!requestBase?.id" jhiTranslate="request.form.header.new">New Request</h1>
+        <h1 *ngIf="requestBase?.id" jhiTranslate="request.form.header.edit">Edit Request</h1>
+        <p *ngIf="selectDraft && !requestBase" jhiTranslate="request.form.header.selectDraft">Select Draft</p>
+    </div>
+    <!--<div class="col-md-8 offset-md-2">
+        <div class="alert alert-danger" *ngIf="authenticationError" >
+            <strong>Failed to sign in!</strong> Please check your credentials and try again.
+        </div>
+    </div>-->
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/form/request-form.component.html]: pls remove if not used

File: podium-gateway/src/main/webapp/app/request/form/request-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        this.principal.identity().then((account) => {
+            console.log('Got user ', account);
+            this.currentUser = account;
+            this.initializeRequestForm();
+        });
+
+        /**
+         * Organisation resolve
+         */
+        console.log('Finding available organisations');
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/form/request-form.component.ts]: pls remove

File: podium-gateway/src/main/webapp/app/request/form/request-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        this.organisationService.findAvailable().map((availableOrganisations) => {
+            // this.availableOrganisations = availableOrganisations;
+        });
+
+        /**
+         * Resolve Tags
+         */
+        console.log('Finding available tags');
+    }
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/form/request-form.component.ts]: pls remove

File: podium-gateway/src/main/webapp/app/request/form/request-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        let uuid = this.currentUser.uuid;
+        this.requestService.findAvailableRequestDrafts(uuid)
+            .subscribe(
+                (requestDrafts) => this.processAvailableDrafts(requestDrafts),
+                (error) => this.onError('Error loading available request drafts.')
+            );
+    }
+
+    registerChangeInFilesUploaded () {
+        console.log('Registering for file upload change.');
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/form/request-form.component.ts]: pls remove

File: podium-gateway/src/main/webapp/app/request/form/request-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            );
+    }
+
+    private postSaveUpdate (requestBase: RequestBase) {
+        console.log('Updated request with {}', requestBase);
+    }
+
+    onError(error) {
+        console.log('An error occurred ', error);
+    }
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/form/request-form.component.ts]: is this still WIP?

File: podium-gateway/src/main/webapp/app/request/form/request-form.component.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    saveRequestDraft () {
+        this.requestService.saveRequestDraft(this.requestBase)
+            .subscribe(
+                (requestBase) => this.postSaveUpdate(requestBase),
+                (error) => this.onError(error)
+            );
+    }
+
+    private postSaveUpdate (requestBase: RequestBase) {
+        console.log('Updated request with {}', requestBase);
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/form/request-form.component.ts]: is this still WIP?

File: podium-gateway/src/main/webapp/app/request/form/request-form.route.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ */
+
+import { Route } from '@angular/router';
+
+import { UserRouteAccessService } from '../../shared';
+import { RequestFormComponent } from './request-form.component';
+
+export const requestFormRoute: Route = {
+  path: 'new',
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/form/request-form.route.ts]: maybe more descriptive something like "new-request" or "create-request"? unless if this is sub path like /request/new

File: podium-gateway/src/main/webapp/app/request/request.module.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import { EnumKeysPipe } from '../shared/pipes/enumKeys';
+
+@NgModule({
+    imports: [
+        PodiumGatewaySharedModule,
+        RouterModule.forRoot(requestRoute, { useHash: true })
+    ],
+    declarations: [
+        RequestFormComponent,
+        EnumKeysPipe
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/request/request.module.ts]: What's enumKeysPipe for?

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+        private jhiLanguageService: JhiLanguageService,
+        private alertService: AlertService,
+        private eventManager: EventManager,
+        private router: Router
+    ) {
+        // this.jhiLanguageService.setLocations(['upload']);
+    }
+
+    ngOnInit() {
+        // this.isSaving = false;
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    public uploader: FileUploader = new FileUploader({ url: URL });
+
+    constructor(
+        public activeModal: NgbActiveModal,
+        private jhiLanguageService: JhiLanguageService,
+        private alertService: AlertService,
+        private eventManager: EventManager,
+        private router: Router
+    ) {
+        // this.jhiLanguageService.setLocations(['upload']);
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+        private route: ActivatedRoute,
+        private uploadPopupService: UploadPopupService
+    ) {}
+
+    ngOnInit() {
+        this.routeSub = this.route.params.subscribe(params => {
+            /*if ( params['uuid'] ) {
+                this.modalRef = this.organisationPopupService
+                    .open(OrganisationDialogComponent, params['uuid']);
+            } else {*/
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+                this.modalRef = this.uploadPopupService
+                    .open(UploadDialogComponent);
+
+                this.modalRef.result
+                    .then((res) => console.log('results ', res))
+                    .catch((err) => console.log('err ', err));
+            // }
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+        this.routeSub = this.route.params.subscribe(params => {
+            /*if ( params['uuid'] ) {
+                this.modalRef = this.organisationPopupService
+                    .open(OrganisationDialogComponent, params['uuid']);
+            } else {*/
+                this.modalRef = this.uploadPopupService
+                    .open(UploadDialogComponent);
+
+                this.modalRef.result
+                    .then((res) => console.log('results ', res))
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.component.ts]: what are these? only logging results?

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+        if (this.isOpen) {
+            return;
+        }
+        this.isOpen = true;
+
+        /*if (uuid) {
+            this.organisationService.findByUuid(uuid).subscribe(organisation => {
+                this.organisationModalRef(component, organisation);
+            });
+        } else { */
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+            return this.uploadModalRef(component);
+        // }
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+        /**
+         * Load request by UUID
+         */
+
+        /**
+         * Else throw error and close dialog on confirm
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+         */
+    }
+
+    uploadModalRef(component: Component): NgbModalRef {
+        let modalRef = this.modalService.open(component, { size: 'lg', backdrop: 'static'});
+        // modalRef.componentInstance.organisation = organisation;
+        modalRef.result.then(result => {
+            console.log(`Closed with: ${result}`);
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts]: pls remove

File: podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+            this.isOpen = false;
+        }, (reason) => {
+            console.log(`Dismissed ${reason}`);
```
Comment (rnugraha) [inline podium-gateway/src/main/webapp/app/shared/upload/upload-dialog.service.ts]: pls remove

**Comment by rnugraha** [comment]
In general would prefer to:

- maintain 120 chars max
- remove logging
- provide services/validator with unit testing (but this may come as separate PR to avoid even bigger PR size)

File: podium-gateway/src/main/webapp/app/request/form/request-form.route.ts (hunk 7 above)
Comment (berndvdveen) [inline podium-gateway/src/main/webapp/app/request/form/request-form.route.ts]: This is a sub path mapping to /request/new

File: podium-gateway/src/main/webapp/app/request/request.module.ts (hunk 8 above)
Comment (berndvdveen) [inline podium-gateway/src/main/webapp/app/request/request.module.ts]: For iterating over the RequestStatus ENUM in the request form.

File: podium-gateway/src/main/java/org/bbmri/podium/domain/enumeration/RequestType.java [hunk 19]
```diff
@@ -0,0 +1,8 @@
+package org.bbmri.podium.domain.enumeration;
+
+/**
+ * The RequestType enumeration.
+ */
+public enum RequestType {
+    DATA,IMAGES,MATERIAL
```
Comment (gijskant) [inline podium-gateway/src/main/java/org/bbmri/podium/domain/enumeration/RequestType.java]: Not sure if this is needed (one biobank may have images, another may have material), but probably it should be a set of request types then? If you want both images and material?

File: podium-gateway/src/main/java/org/bbmri/podium/service/dto/OrganisationDTO.java [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+
+import javax.validation.constraints.*;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * A DTO for the Organisation entity.
+ */
+public class OrganisationDTO implements Serializable {
```
Comment (gijskant) [inline podium-gateway/src/main/java/org/bbmri/podium/service/dto/OrganisationDTO.java]: Why not in podium-common?

**Comment by rnugraha** [comment]
I have no anymore comments

File: podium-gateway/src/main/java/org/bbmri/podium/domain/enumeration/RequestType.java (hunk 19 above)
Comment (berndvdveen) [inline podium-gateway/src/main/java/org/bbmri/podium/domain/enumeration/RequestType.java]: Good point. Ill make it a set in RequestDetail and add a element collection table

Comment (berndvdveen) [inline podium-gateway/src/main/java/org/bbmri/podium/domain/enumeration/RequestType.java]: @gijskant Please see 167f327

File: podium-gateway/src/main/java/org/bbmri/podium/service/dto/OrganisationDTO.java (hunk 20 above)
Comment (berndvdveen) [inline podium-gateway/src/main/java/org/bbmri/podium/service/dto/OrganisationDTO.java]: Migrated to common

**Pull request merged**